### PR TITLE
Fix compilation error when using memory database

### PR DESCRIPTION
**Pull request by danielabrozzoni**
This commit also adds the testing for memory database and key value
database in CI

#### All Submissions:

* [x] I've signed all my commits
* [x] I followed the [contribution guidelines](https://github.com/bitcoindevkit/bdk-cli/blob/master/CONTRIBUTING.md)
* [x] I ran `cargo fmt` and `cargo clippy` before committing

#### Bugfixes:

* [ ] This pull request breaks the existing API
* [x] I've added tests to reproduce the issue which are now passing
* [ ] I'm linking the issue being fixed by this PR

### CODE REVIEW

**Review by afilini — APPROVED** [review]
ACK d8759af75cfe24b4138ef4ea8a51d1a2c3644e4e

File: src/utils.rs [hunk 1]
```diff
@@ -183,6 +183,9 @@ pub(crate) fn open_database(wallet_opts: &WalletOpts) -> Result<AnyDatabase, Err
             .into_string()
             .expect("path string"),
     });
+    #[cfg(not(any(feature = "key-value-bd", feature = "sqlite-db")))]
```
Comment (notmandatory) [inline src/utils.rs:186]: A little typo to fix here:
```suggestion
    #[cfg(not(any(feature = "key-value-db", feature = "sqlite-db")))]
```

**Review by notmandatory — CHANGES_REQUESTED** [review]
Looks good, just a little typo to fix.

File: src/utils.rs (hunk 1 above)
Comment (notmandatory) [reply src/utils.rs:186]: PS. I updated the github checks to match the new workflow matrix.

Comment (danielabrozzoni) [reply src/utils.rs:186]: Ohhh this would have been horrible to debug, thanks!

**Review by notmandatory — APPROVED** [review]

**Review by notmandatory — APPROVED** [review]
ACK faa461d3fc8d20cc033859555017792017e87461

**Pull request merged**